### PR TITLE
Add/type annotations/model vision

### PR DESCRIPTION
**Pull request by johnnv1**
I added type annotations for  Beit, ViT and Deit models at PyTorch as described in #16059 

@Rocketknight1 

fix the PR #16092 

Also, the issue with `make fix-copies` for ViT_mae, maybe I should open an issue?

For this PR, I just remove the type annotation of config parameter that cause the problem

### CODE REVIEW

**Comment by Rocketknight1** [comment]
This looks really good! Good job handling the more intricate `Tuple` annotations as well as forward references well - this is a lot bigger and more challenging than most of the other type hint PRs, but it looks like you did it well. 

I have one final small issue - types should only be `Optional` when `None` is an allowable option. It looked like there were some cases when `Optional[bool]` was used for boolean arguments to internal methods which have to be either `True` or `False` - these should just be `bool`. Can you double-check that? Of course, when `None` is allowable or the default option, then `Optional[bool]` is correct. I'm not 100% certain on this one (maybe `None` really is allowable in all of those cases!), so if you think all your annotations were correct, let me know.

**Comment by johnnv1** [comment]
Hi, i have kept `Optional[bool]` just to keep everything equals. But I will take a look, if it can be changed to "bool", I will update it and let you know.

**Comment by johnnv1** [comment]
@Rocketknight1 i have changed to just `bool`, both cases work, but just `bool` seems more correct and clean for me.

When has `None` as default, the code will be `a if x is not None else b`, and with `False`/`True` will be just `a if x else b`

**Comment by Rocketknight1** [comment]
Sounds good, reviewing now!

File: src/transformers/models/beit/modeling_beit.py [hunk 1]
```diff
@@ -904,7 +925,15 @@ class BeitConvModule(nn.Module):
     Based on OpenMMLab's implementation, found in https://github.com/open-mmlab/mmsegmentation.
     """
 
-    def __init__(self, in_channels, out_channels, kernel_size, padding=0, bias=False, dilation=1):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Tuple[int, int]],
+        padding: Union[int, Tuple[int, int], str] = 0,
```
Comment (Rocketknight1) [inline src/transformers/models/beit/modeling_beit.py:933]: This is absolutely heroic type annotation

**Review by Rocketknight1 — APPROVED** [review]
This is a fantastic (and massive!) PR. I read through the whole thing and couldn't find any wrong types, so I'm happy to accept it as-is. Thank you!

**Pull request merged**